packages/models/src/agreement/agreement.ts

```diff
@@ -39,6 +40,7 @@ export type AgreementDocument = z.infer<typeof AgreementDocument>;
 
 export const AgreementStamp = z.object({
   who: UserId,
+  delegationId: DelegationId.optional(),
```

Comment:
Same comment about converters left for `rejectionReasons` also for this new optional field